src/test/crimson/test_backfill_facades.h

```diff
@@ -0,0 +1,39 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "crimson/osd/backfill_state.h"
+
+namespace crimson::osd {
+
+// PeeringFacade -- a facade (in the GoF-defined meaning) simplifying
+// the interface of PeeringState. The motivation is to have an inventory
+// of behaviour that must be provided by a unit test's mock.
+struct BackfillState::PeeringFacade {
```

Comment:
why are we defining the facades here, in the testing code, and not as a separate .h files in the "real" code?

Reply:
i think these are interfaces for stubs which mimics the behavior of their counterparts defined in `src/crimson/osd/backfill_facades.h`.  this is probably not the most elegant way to do the unit test. maybe we should indeed define an interface of these two façades, and plug their implementations at runtime. the interfaces can be shared by both tests and production code.

could leave this in a follow-up PR though.

Reply:
Exactly, this is another implementation of the **implicit** interface the facade in `src/crimson/osd/backfill_facades.h` also conforms to. The motivation was to avoid dynamic polymorphism and CRTP. After awhile I perceive this as a premature optimization. Do we have a proof from a profiler showing the `virtual`-driven overhead makes a problem? Let's not sacrifice explicitness for a vague performance benefit.

I will introduce the explicit interface in a follow-up PR.